packages/adapter-vercel/index.js

```diff
@@ -37,6 +37,16 @@ const get_default_runtime = () => {
 // https://vercel.com/docs/functions/edge-functions/edge-runtime#compatible-node.js-modules
 const compatible_node_modules = ['async_hooks', 'events', 'buffer', 'assert', 'util'];
 
+const [major, minor] = VERSION.split('.').map(Number);
+const can_use_middleware = major > 2 || (major === 2 && minor > 17);
```

Comment:
Note that this is currently always `false` because we're not at 2.18 yet (we will if this is released); so if you want to test this out you gotta adjust this.